ocs_ci/deployment/cnv.py

```diff
@@ -648,3 +648,90 @@ def disable_multicluster_engine(self):
             logger.error(f"Failed to disable multicluster engine\n{cmd_res.stderr}")
             return
         logger.info(cmd_res.stdout.decode("utf-8").splitlines())
+
+    def remove_hyperconverged(self):
+        """
+        Remove HyperConverged CR
+
+        """
+        hyperconverged_obj = OCP(
+            kind=constants.HYPERCONVERGED,
+            resource_name=constants.KUBEVIRT_HYPERCONVERGED,
+            namespace=self.namespace,
+        )
+        hyperconverged_obj.delete(resource_name=constants.KUBEVIRT_HYPERCONVERGED)
+        logger.info(
+            f"Deleted {constants.HYPERCONVERGED} {constants.KUBEVIRT_HYPERCONVERGED}"
+        )
+
+    def remove_cnv_subscription(self):
+        """
+        Remove CNV subscription
+
+        """
+        cnv_sub = OCP(
+            kind=constants.SUBSCRIPTION,
+            resource_name=constants.KUBEVIRT_HYPERCONVERGED,
+            namespace=self.namespace,
+        )
+        cnv_sub.delete(resource_name=constants.KUBEVIRT_HYPERCONVERGED)
+        logger.info(f"Deleted subscription {constants.KUBEVIRT_HYPERCONVERGED}")
+
+    def remove_cnv_csv(self):
+        """
+        Remove CNV ClusterServiceVersion
+
+        """
+        cnv_csv = OCP(
+            kind=constants.CLUSTER_SERVICE_VERSION,
+            selector=constants.CNV_SELECTOR,
+            namespace=self.namespace,
+        )
+        cnv_csv.delete(resource_name=cnv_csv.get()["items"][0]["metadata"]["name"])
+        logger.info(f"Deleted ClusterServiceVersion {constants.CNV_OPERATORNAME}")
+
+    def remove_crds(self):
+        """
+        Remove openshift virtualization CRDs
+
+        """
+        OCP().exec_oc_cmd(
+            command=f"delete crd -n {self.namespace} -l {constants.CNV_SELECTOR}"
+        )
+        logger.info("Deleted all the openshift virtualization CRDs")
+
+    def remove_namespace(self):
+        """
+        Remove openshift virtualization namespace
+
+        """
+        cnv_namespace = OCP(
+            kind=constants.NAMESPACE, resource_name=constants.CNV_NAMESPACE
+        )
+        cnv_namespace.delete(resource_name=constants.CNV_NAMESPACE)
+        logger.info(f"Deleted the namespace {constants.CNV_NAMESPACE}")
+
+    def cleanup_cnv(self, check_cnv_installed=False):
+        """
+        Uninstall CNV deployment
```

Comment:
You need to also check if there are any running virtual machines and virtual machine instances on the cluster. You cannot uninstall OpenShift Virtualization while its workloads remain on the cluster.